subsys/bluetooth/host/keys_br.c

```diff
@@ -52,31 +59,66 @@ struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr)
 	}
 
 	key = bt_keys_find_link_key(BT_ADDR_ANY);
+#if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (!key) {
+		int i;
+
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			struct bt_keys_link_key *current = &key_pool[i];
+
+			if (current->aging_counter < key->aging_counter) {
```

Comment:
This is a NULL pointer on the first iteration of the loop AFAICT.

I suppose the intended use was to assign to have `key = &key_pool[0];` before the loop begins, otherwise I don't understand why the loops is starting at 1 either.